bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/UdpListener.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.yamahamusiccast.internal;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link UdpListener} is responsible for receiving
+ * UDP braodcasts from the Yamaha device/model.
+ *
+ * @author Lennert Coopman - Initial contribution
+ */
+@NonNullByDefault
+public class UdpListener extends Thread {
```

Comment:
You should start this thread via the framework to name and configure it correctly. You could make this a `Runnable` and use `Executors.newSingleThreadExecutor(new NamedThreadFactory("binding-xyz")` to start the thread.

Reply:
This new to me. Is there another binding which I can use as an example?

Reply:
Sure: https://github.com/openhab/openhab-addons/search?q=NamedThreadFactory

Reply:
I have added a Runnable to the BridgeHandler. This one is called via Executors. As this is new to me, I took a look at the Miele Binding.
The old code is still there but the function is not started anymore. The UDP events are caught and reflected in OH.
If this is OK, I will clean up the code. If not, the old code is currently my fallback.
As far as I can see, this is the last remark I had to check/modify, I will request a new review.

Reply:
Your changes look good. I made a remark.

The log messages don't need to be prefixed with YXC as the class name is logged, too by the logging system.

Reply:
Thanks. Removed YXC from all log messages.
Will now work on cleaning up the code, compile and test it. Then a push will follow.